src/DefaultBuilder/src/BootstrapHostBuilder.cs

```diff
@@ -114,26 +105,59 @@ public void RunDefaultCallbacks(HostBuilder innerBuilder)
                 configureHostAction(_configuration);
             }
 
-            // Configuration doesn't auto-update during the bootstrap phase to reduce I/O,
-            // but we do need to update between host and app configuration so the right environment is used.
-            _environment.ApplyConfigurationSettings(_configuration);
+            // This is the hosting environment based on configuration we've seen so far.
+            var hostingEnvironment = new HostingEnvironment()
+            {
+                ApplicationName = _configuration[HostDefaults.ApplicationKey],
+                EnvironmentName = _configuration[HostDefaults.EnvironmentKey] ?? Environments.Production,
```

Comment:
Is it OK to fallback to `Production` here?

Reply:
Yes, we default to production in absence of other configuration. This logic also mimics the default host https://github.com/dotnet/runtime/blob/cf52b7e38d28b7e28358cee052effe0ed47efb2e/src/libraries/Microsoft.Extensions.Hosting/src/HostBuilder.cs#L180-L196